doc/endianness.md

```diff
@@ -0,0 +1,29 @@
+# Endianness
+
+## default endianness
+
+SPDM 1.0 defines the default endianness is little endian.
+
+libspdm follows that. Assuming the libspdm compiler is using little endian, no swap is required.
+
+## endianness of digital signature
+
+SPDM 1.2 defines the endianness of digital signature for RSA, ECDSA, SM2_DSA, and EdDSA.
+
+libspdm follows that. Because the definition aligns with existing crypto library such as openssl and mbedtls, no swap is required.
+
+SPDM 1.0 and 1.1 are not very clear on the digital signature for RSA and ECDSA.
+
+**libspdm uses big endian** for RSA signature and ECDSA, which algins with SPDM 1.2 and existing crypto library. No swap is required.
+
+## endianness of key exchange data
+
+SPDM 1.1/1.2 defines the endianness of key exchange data for FFDHE, ECDHE, and SM2_KeyExchange.
+
+libspdm follows that. Because the definition aligns with existing crypto library such as openssl and mbedtls, no swap is required.
+
+## endianness of AEAD IV
+
+Secured SPDM 1.0/1.1 are not very clear on how to extend 64bit sequence number and XOR with the IV derived from SPDM key schedule.
+
+**libspdm uses little endian** for the sequence number, which algins with default endianness. No swap is required.
```

Comment:
A libspdm Requester still needs to support devices that encode the sequence number as big-endian, so that will be configurable. The default endianness can be little.

Reply:
That is future task.
The current document should just describe the current behavior.